src/applications/detection/main.cpp

```diff
@@ -51,236 +52,340 @@ using boost::timer;
 using namespace boost::gil;
 namespace bfs = boost::filesystem;
 
-void detection( std::size_t frameId,
-                int         pipeId,
-                const cv::Mat & src,
-                const cctag::Parameters & params,
-                const cctag::CCTagMarkersBank & bank,
-                std::ostream & output,
-                std::string debugFileName = "")
+/**
+ * @brief Check if a string is an integer number.
+ * 
+ * @param[in] s The string to check.
+ * @return Return true if the string is an integer number
+ */
+bool isInteger(std::string s)
 {
-  
-    if (debugFileName == "") {
-      debugFileName = "00000";
+  std::regex e("^-?\\d+");
+  return std::regex_match(s, e);
+}
+
+/**
+ * @brief Draw the detected marker int the given image. The markers are drawn as a
+ * circle centered in the center of the marker and with its id. It draws the 
+ * well identified markers in green, the unknown / badly detected markers in red.
+ * 
+ * @param[in] markers The list of markers to draw.
+ * @param[out] image The image in which to draw the markers.
+ */
+void drawMarkers(const boost::ptr_list<CCTag> &markers, cv::Mat &image)
+{
+  BOOST_FOREACH(const cctag::CCTag & marker, markers)
+  {
+    const cv::Point center = cv::Point(marker.x(), marker.y());
+    const int radius = 10;
+    if(marker.getStatus() == 1)
```

Comment:
yes i didn't see
https://github.com/poparteu/CCTag/blob/develop/src/cctag/CCTag.hpp#L416

Maybe make it an enum rather than a namespace + static var?